### PR TITLE
preliminaries for user switching and also cleanups

### PR DESCRIPTION
**Pull request by maxtaco**
- get rid of `context.Context` inside of `GlobalContext`
- pass `context.Context` through to some methods on `GlobalContext`

### CODE REVIEW

File: go/libkb/globals.go [hunk 1]
```diff
@@ -280,23 +263,48 @@ func (g *GlobalContext) simulateServiceRestart() {
...
+	}
+	delete(g.switchedUsers, username)
+}
+
+func (g *GlobalContext) logoutWithSecretKill(mctx MetaContext, killSecrets bool) (err error) {
+
+	ctx := mctx.Ctx()
+	defer g.switchUserMu.Acquire(NewMetaContext(ctx, g), "Logout")()
+
+	username := g.ActiveDevice.Username(mctx)
```
Comment (maxtaco) [inline go/libkb/globals.go]: Less racey than digging around in the environment.

File: go/libkb/globals.go [hunk 2]
```diff
@@ -142,10 +142,9 @@ type GlobalContext struct {
 	// It is threadsafe to call methods on ActiveDevice which will always be non-nil.
 	// But don't access its members directly. If you're going to be changing out the
 	// user (and resetting the ActiveDevice), then you should hold the switchUserMu
-	switchUserMu *VerboseLock
-	ActiveDevice *ActiveDevice
-
-	NetContext context.Context
+	switchUserMu  *VerboseLock
+	ActiveDevice  *ActiveDevice
+	switchedUsers map[NormalizedUsername]bool
```
Comment (maxtaco) [inline go/libkb/globals.go]: users who have been "switched" who, in the old regime, were just logged out

File: go/libkb/globals.go [hunk 3]
```diff
@@ -280,23 +263,53 @@ func (g *GlobalContext) simulateServiceRestart() {
...
 func (g *GlobalContext) Logout(ctx context.Context) (err error) {
 	mctx := NewMetaContext(ctx, g).WithLogTag("LOGOUT")
-	ctx = mctx.Ctx()
-
 	defer mctx.Trace("GlobalContext#Logout", func() error { return err })()
+	return g.logoutWithSecretKill(mctx, true)
+}
 
-	defer g.switchUserMu.Acquire(NewMetaContext(ctx, g), "Logout")()
+func (g *GlobalContext) switchDoLogout(mctx MetaContext) (err error) {
```
Comment (patrickxb) [inline go/libkb/globals.go]: is this called by anything?

Comment (maxtaco) [reply go/libkb/globals.go]: in the next pr it is....

**Pull request merged**